src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -204,30 +192,40 @@
                        "listed in PuppetDB's certificate-whitelist file?")
                   ssl-client-cn))))))
 
-(defn shutdown-mq-broker
+(defn shutdown-mq
   "Explicitly shut down the queue `broker`"
-  [{:keys [broker]}]
+  [{:keys [broker mq-factory mq-connection]}]
   (when broker
-    (log/info "Shutting down message broker.")
-    ;; Stop the mq the old-fashioned way
+    (log/info "Shutting down the messsage queues.")
+    (.close mq-connection)
+    (.close mq-factory)
     (mq/stop-broker! broker)))
 
+(defn shutdown-updater
+  [context]
+  (when-let [updater (context :updater)]
+    (log/info "Shutting down updater thread.")
+    (future-cancel updater)
+    (let [timeout (Object.)]
+      (try
+        (when (= timeout (deref updater 3000 timeout))
+          (log/error "waited for updater thread for 3s but it didn't finish"))
```

Comment:
Like I said in the channel I don't think we necessarily want to log an error if they don't connect since it doesn't affect puppetdb, and it may be the case for some users that they aren't connected to the internet (which we've heard can often be the case for our services).